cumulus/polkadot-parachain/src/service.rs

```diff
@@ -686,85 +683,6 @@ where
 	Ok(BasicQueue::new(verifier, Box::new(block_import), None, &spawner, registry))
 }
 
-/// Start an aura powered parachain node. Some system chains use this.
-pub async fn start_generic_aura_node(
```

Comment:
Isn't this part of the public API, `start_generic_aura_node` what if other parachains want to bump the crate version but they don't care about upgrading to the lookahead,  wouldn't they get errors  or am I missing something ?

Reply:
Well, I've set a version bump to "major" in the prdoc, so the breaking change is okay. Here, we're effectively deprecating the generic collator in favor of the lookahead one. This makes sense, as the lookahead collator can do everything the generic collator can do, with some additional functionality.

I doubt if anyone really uses `polkadot-parachain` to run any non-Parity parachains, but if they do, it's not that big an effort to switch to the lookahead, basically, it's just calling another function.

So it seems sound to me, but thanks for bringing up the concern, maybe we'll get other opinions on that.

Reply:
I would go with some soft deprecation like this `#[deprecated = "why is deprecated ....."]`, it just makes easier for people to upgrade, it also gives us an entry point to notify them what to do to stop using this function. The alternative is they get a compilation error when they run `cargo  update`

Reply:
But `cargo update` wouldn't update through a major version afaik?

Reply:
>But cargo update wouldn't update through a major version afaik?

That's true, I was trying to just give an example of unrelated mindless update people might do and they might give up because of compilation errors.

Anyways, I don't have data to back up that this might break other peoples flows, however I do think it is cheap to keep it around for a few releases as deprecated and that would be the less risky alternative since I do see how it would hurt us to keep it around.

Reply:
Okay, now I found out that it's not a public interface indeed, it's declared `pub`, but it lives in a private module and is never reexported. Are you still in favor of keeping it for a while? :upside_down_face:

Reply:
If it is not exported outside, not need to keep it we can restore it from git if we need.